src/hostnet/lib/slirp.ml

```diff
@@ -152,9 +156,9 @@ let connect x peer_ip local_ip =
             );
             Tcpip_stack.listen_tcpv4_flow s ~on_flow_arrival:(
               fun ~src:(src_ip, src_port) ~dst:(dst_ip, dst_port) ->
-
                 let for_us src_ip = Ipaddr.V4.compare src_ip local_ip == 0 in
-                ( if for_us src_ip && src_port = 53 then begin
+                let for_dns src_ip = for_us src_ip || Ipaddr.V4.compare src_ip extra_dns_ip == 0 in
+                ( if for_dns src_ip && src_port = 53 then begin
                     Resolv_conf.get () (* re-read /etc/resolv.conf *)
                     >>= function
                     | (Ipaddr.V4 ip, port) :: _  -> Lwt.return (ip, port)
```

Comment:
Ah yes, I punted on that because the current code wasn't doing round-robin yet either and then GC'd the `XXX` I'd left in a fit of cleanliness...

I think what I'd want to do is refactor the server selection stuff which I added to `Dns_forward.input` into a new helper which could be called here too to select the upstream server to go to.